include/boost/geometry/algorithms/detail/disjoint/multipoint_geometry.hpp

```diff
@@ -75,12 +76,12 @@ class multipoint_multipoint
         {}
 
         template <typename Point>
-        inline bool apply(Point const& point) const
+        inline bool operator()(Point const& point) const
         {
-            return !std::binary_search(m_first,
-                                       m_last,
-                                       point,
-                                       static_cast<base_type const&>(*this));
+            return std::binary_search(m_first,
+                                      m_last,
+                                      point,
+                                      static_cast<base_type const&>(*this));
```

Comment:
You have changed the behavior of predicates and left the old names. E.g. in this case this predicate no longer is `unary_disjoint_predicate` it's `unary_intersects_predicate`.